theme/lumo/vaadin-custom-field-styles.html

```diff
@@ -43,20 +43,27 @@
         height: calc(0.4em - var(--lumo-space-xs));
       }
 
-      :host([focused]:not([disabled])) [part="label"] {
+      :host([focused]:not([readonly]):not([disabled])) [part="label"] {
         color: var(--lumo-primary-text-color);
       }
 
-      :host(:hover:not([disabled]):not([focused])) [part="label"] {
+      :host(:hover:not([readonly]):not([disabled]):not([focused])) [part="label"] {
         color: var(--lumo-body-text-color);
       }
 
       /* Touch device adjustment */
       @media (pointer: coarse) {
-        :host(:hover:not([disabled]):not([focused])) [part="label"] {
+        :host(:hover:not([readonly]):not([disabled]):not([focused])) [part="label"] {
           color: var(--lumo-secondary-text-color);
         }
       }
+
+      /* Disabled style */
+
+      :host([disabled]) [part="label"] {
+        color: var(--lumo-disabled-text-color);
+        -webkit-text-fill-color: var(--lumo-disabled-text-color);
```

Comment:
This line is copied from text field label styles and only needed for this disabled style. I checked that it's actually needed atm. otherwise the required indicator would not be gray when disabled (this overrides the `color` on the inner `::after` element and it seems to work at least on Chrome, Edge 18 and Firefox).

This could probably be refactor to not need a vendor property, but now it's at least consistent with text field styles.